sdk/keyvault/azure-security-keyvault-administration/src/samples/java/com.azure.security.keyvault.administration/CreateRoleAssignmentsForDifferentScopes.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.administration;
+
+import com.azure.identity.DefaultAzureCredentialBuilder;
+import com.azure.security.keyvault.administration.models.KeyVaultRoleAssignment;
+import com.azure.security.keyvault.administration.models.KeyVaultRoleAssignmentProperties;
+import com.azure.security.keyvault.administration.models.KeyVaultRoleAssignmentScope;
+
+/**
+ * This sample demonstrates how to create role assignments in the key vault for different scopes.
+ */
+public class CreateRoleAssignmentsForDifferentScopes {
+    /**
+     * Authenticates with the key vault and shows how to create role assignments in the key vault for different scopes.
+     *
+     * @param args Unused. Arguments to the program.
+     * @throws IllegalArgumentException when an invalid key vault URL is passed.
+     */
+    public static void main(String[] args) {
+        // Instantiate an KeyVaultAccessControlClient that will be used to call the service. Notice that the client is
+        // using default Azure credentials. To make default credentials work, ensure that environment variables
+        // 'AZURE_CLIENT_ID', 'AZURE_CLIENT_KEY' and 'AZURE_TENANT_ID' are set with the service principal credentials.
+        KeyVaultAccessControlClient accessControlClient = new KeyVaultAccessControlClientBuilder()
+            .vaultUrl("https://{YOUR_VAULT_NAME}.vault.azure.net")
+            .credential(new DefaultAzureCredentialBuilder().build())
+            .buildClient();
+
+        // By default role assignments apply to the global scope. It is also possible to be more specific by applying an
+        // assignment to the all keys scope or a specific KeyVaultKey.
+
+        // Let's assign a role to a service principal so that it applies to all keys. To do this we'll need a service
+        // principal object ID and a role definition ID. A role definition ID can be obtained from the 'id' property of
+        // one of the role definitions returned from listRoleAssignments().
```

Comment:
listRoleDefinitions. They can also get this with the azure CLI, which you've used before:
```bash
az keyvault role definition list --hsm-name <name>
```